### ISSUE TITLE
ARM Linux installers (Raspberry Pi)

### ISSUE THREAD

**Issue by jflamy** [shannah/jdeploy#121]
Just looking at the documentation: there is no mention of a Linux install for Raspberry Pi OS (.deb) or other ARM-based computers.
Is that present and not documented?

**Comment by shannah** [comment]
This has been on my todo list for some time.  The main blocker (last time I checked) was that github actions, which I use to automate builds and releases, doesn't have ARM linux runners, so either I need to set up my own runner (which isn't impossible - just haven't had time to look into it), or I need to set up a cross-compile build flow on linux (also not impossible, but haven't had time to look into it).

If you have any experience in either of these areas (self-hosted github runners -- preferred solution), or cross compiling for ARM linux on x86, I'd welcome help on this issue.

**Comment by jflamy** [comment]
My application already has a fairly convoluted maven build that I run under Azure devops, I have never used github actions, that is one of the things that are currently stopping me from jumping in,
So no, I don't have any github experience -- and in fact having a release script would be useful for other types of build (in-house or other clouds).
My setup is a dev repository with manually triggered builds that send the releases to two separate release repos, one for stable releases, one for pre-releases.  I will also need to figure out how jdeploy behaves when branches have been used.

**Comment by shannah** [comment]
Are you using jDeploy to deploy a desktop app, or a CLI app?

**Comment by jflamy** [comment]
A CLI that wants to be a desktop (https://owlcms.github.io/owlcms4/#/index scroll down for screenshots)
 
It's a web app packaged as a uberjar.  It is deployed on a "main" laptop on a LAN, where it runs an embedded web server (like SpringBoot would). It also starts a browser window where a user can interact.   Other laptops on the same LAN can then connect to the main laptop and open their own browser.

So what I am attempting to do is have something less crude than a Java console that starts the web server. Because on the main laptop it is easy to just kill that window -- and on Windows if you do that there are no shutdown hooks, it's like a "kill -9". 

I would like to have my main Java be a tiny/prettier Swing or JavaFX desktop app that forks off the web server and controls it (that part is already there, that's what the CLI version does).

jdeploy would be nice -- I would need to figure out how to have multiple branches at any given time, as there are multiple stable versions out at any given time (like Windows 10 and Windows 11 both getting fixes).

Feel free to move this to a discussion.

**Comment by FDelporte** [comment]
It seems Linux ARM runners are now available for GitHub Actions: https://github.blog/2024-06-03-arm64-on-github-actions-powering-faster-more-efficient-build-systems/

Would indeed be great to have installers for RPi and other Linux ARM platforms...

**Comment by shannah** [comment]
Yeah.  I don't think this is available to me yet.   I think what I'm going to do is set up a self-hosted runner on a raspberry pi or maybe use QEMU.

With jDeploy all the native stuff is precompiled, so you don't actually need to have arm64 available when building your app.  I just need to precompile the launcher for that platform - so if I were desperate, I could just manually compile it for arm64 and add this to the jdeploy resources.  But I'd prefer to get it automated in a pipeline.

**Comment by shannah** [comment]
This has been added as part of v5.0

Currently they are "off" by default.

The jDeploy desktop app now has a Download Page tab that allows you to configure the platforms that you want to deploy for.  "Default" is all of the existing platforms that were on the download page.  "All" will be all platforms that jDeploy supports.
Or you can select a custom set.

<img width="1029" height="795" alt="Image" src="https://github.com/user-attachments/assets/000da384-2ced-40c4-9f2f-12382a230e7b" />

If you want to deploy ARM builds of your app for Windows and Linux, you should either select "All", or specifically check them in custom.

You can preview your download page with arm runners by adding ?preview=1 to the URL.  
e.g. https://www.jdeploy.com/~snapcodejava?preview=1

<img width="1672" height="1083" alt="Image" src="https://github.com/user-attachments/assets/29e09d96-2154-42da-a0f0-df7c3aacadff" />

5.0 is not the default in the github action yet.  I will likely update that this weekend.  In the mean-time you can add `"jdeploy_version": "5.0.1"` to the parameters of the jdeploy action if you want the arm builds.
